core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkerInternalQueue.scala

```diff
@@ -45,15 +52,49 @@ trait WorkerInternalQueue {
 
   private val controlQueue = lbmq.getSubQueue(CONTROL_QUEUE)
 
+  private var inputToCredits = new mutable.HashMap[ActorVirtualIdentity, Int]()
```

Comment:
maybe add a comment here saying the value of this map is number of tuples (rather than batches)

Reply:
inputToCredits